### PR TITLE
Update Mapbox cassettes to support city-only geocoding queries

### LINKED ISSUE #173: Mapbox (Carmen GeoJSON format) is parsed incorrectly
Currently the Mapbox implementation sets address attributes based on the `features[].context`, but that object only contains _parent_ information about a `feature` according to the [Carmen GeoJSON](https://github.com/mapbox/carmen/blob/master/carmen-geojson.md) format.

```
1.9.3-p448 :031 > foo = Geokit::Geocoders::MapboxGeocoder.geocode('Los Angeles, CA, United States')
 => Provider: mapbox
Street:
City:
State: California
Zip:
Latitude: 34.0522
Longitude: -118.2437
Country:
Success: true
1.9.3-p448 :032 > foo.city
 => nil
```

I am looking into how to best parse the Carmen GeoJSON format now.

### PR DESCRIPTION
**Pull request by scudco**
Fixes #173 

Note that mapbox has removed any notion of `country_code` from their responses :(

### CODE REVIEW

File: lib/geokit/geocoders/mapbox.rb [hunk 1]
```diff
@@ -68,6 +67,9 @@ def self.set_address_components(result_json, loc)
         if result_json["place_name"]
           loc.full_address = result_json["place_name"]
         end
+        if !loc.city && result_json["id"] =~ /^city\./
+          loc.city = result_json['text']
```
Comment (scudco) [inline lib/geokit/geocoders/mapbox.rb]: Got it :+1:

**Comment by sdhull** [comment]
:+1:

**Comment by rgarver** [comment]
:+1:

**Comment by mnoack** [comment]
Sorry for the late response.

**Pull request merged**